### DISCUSSION TITLE
The sampling frequency does not match the recording interval, Continuous build 3b39742.

### DISCUSSION THREAD

**Discussion by Terracotta-62** [artisan-roaster-scope/artisan#1168] [General]
![82285a66e10f5660955162b9f2a55d3](https://github.com/artisan-roaster-scope/artisan/assets/116416332/54599c2b-07e8-42c7-8d4a-a3701fa30cfc)

**Comment by MAKOMO** [comment]
Thanks for your report. I cannot reproduce this. Note that sampling at a 1sec rate means that every second a request is send to the connected devices asking for data. If that data arrives in time it will be registered. In case a device fails to deliver data at the speed the requests are send the data (table) usually shows irregular intervals. The data is always registered at the time it is received.

What is your hardware setup? Is it possible that your hardware cannot process requests and deliver data at a 1sec rate?

**Comment by Terracotta-62** [comment]
Fuji PUMA

Ramp soak
Motorized valve control
Extended input
Extended output
Cascade control without external wiring
Fast control within 100 ms
More functions for the parameter loader

**Reply by MAKOMO** [reply]
@Terracotta-62 : So you are using the Fuji device or are you using a custom MODBUS setup using the MODBUS meter device? Do you see this problem only with the current Continuous build or also with the any previous release? Any communication errors in the log file?

Could you please attach your artisan-settings.aset (rename to artisan-settings.txt) here as well as the artisan.log file (click the plus icon in the upper left corner while holding the ALT/OPTION key. This should open an email with the log file attach. Also this file you need to rename to artisan.log.txt to be able to drag-and-drop it here on Github. THANKS!

**Reply by Terracotta-62** [reply]
v2.8.3-RC is perfect, thanks!

**Comment by smaca69** [comment]
I have the same issue and I have the sampling rate set to 1sec.  I'm using
a UG22C and it shouldn't have an issue with delivering data at 1sec
intervals

[image: image.png]

**Reply by MAKOMO** [reply]
@smaca69 : which version of Artisan are you using? Did you experience the issue with any Artisan version before?

Could you please attach your artisan-settings.aset (rename to artisan-settings.txt) here as well as the artisan.log file (click the plus icon in the upper left corner while holding the ALT/OPTION key. This should open an email with the log file attach. Also this file you need to rename to artisan.log.txt to be able to drag-and-drop it here on Github. THANKS!

**Comment by dfunni** [comment]
I am having the same problem in v2.10.0 on Raspberry Pi OS Bookworm. Setting the sample period to 1.00s gives an actual sampling period of 2.000 +/- 0.001 s (the precision is actually pretty impressive). I've tried adjusting the sampling period lower, from 0.25 to 0.75, all of which result in an actual interval of 1.500 +/- 0.001s. I have also confirmed that Artisan is only sending a single READ serial command (set up for TC4) each interval.

Settings and an example log file attached:
[artisan-settings.txt](https://github.com/artisan-roaster-scope/artisan/files/13706876/artisan-settings.txt)
[23-12-18_0933.alog.txt](https://github.com/artisan-roaster-scope/artisan/files/13706877/23-12-18_0933.alog.txt)

**Reply by MAKOMO** [reply]
What happens if you put your serial timeout from 0.8 down to 0.4sec (menu Config >> Port, first tab "ET/BT")?

**Reply by dfunni** [reply]
Thanks for the tip, I was able to fix the issue by adjusting the serial timeout as you suggested!
